### ISSUE TITLE
[469b] Text pasted into UEd's Script Editor is unformatted and all on one line

### ISSUE THREAD

**Issue by OjitroC** [OldUnreal/UnrealTournamentPatches#354]
Pasting text like this

function PreSetMovement()
{
bCanJump = true;
bCanWalk = true;
bCanSwim = false;
bCanFly = true;
MinHitWall = -0.6;
bCanOpenDoors = true;
bCanDoSpecial = true;
bCanDuck = true;
}

copied from a plain text .uc file into UEd's Script Editor in 469b results in the pasted text all being on one line - doing the same thing in 436 results in the text being pasted correctly with line breaks.

To reproduce
extract the script of a ScriptedPawn (for example) as an .uc file using UTPT;
copy part of the script
open UEd and open the Script Editor
paste the copied text in the Script Editor

Result that I get : the text is all on one line
Expected result : the text is correctly formatted with line breaks etc

My plain text .uc file is in Western (Mac OS Roman) plain text encoding.  Saving that file in other encodings like Unicode (UTF - 16 or UTF - 8) results in the same outcome.  Copying and pasting the same text from a webpage results in the text being correctly formatted in the Script Editor (that is, copying the text from here - https://ut99.org/viewtopic.php?f=1&t=14127&p=128707#p128660)

I should point out that I am using an iMac and running UT (and hence the Editor) in a WineSkin wrapped with a Windows version equivalent to XP.

**Comment by SeriousBuggie** [comment]
Mostly important specify type of used line breaks.
Look like now it is "\r". When UnrealEd except "\r\n".

**Comment by OjitroC** [comment]
I've no idea what type of line break there is in the text files - how do I find out?

I can paste the same text into the UEd in 436 without any problems so there is something different in the 469b Script Editor - which is somewhat annoying.

**Comment by OjitroC** [comment]
I see from issue 353 that the Script Editor has become a RTF editor, unlike the 436 Editor which accepts both RTF text and plain text.  Pasting 'stuff' from a RTF file into the 469b Script Editor results in it appearing correctly (of course). 

I've always used plain text as that is what the .int files are (or at least that's how they appear to me).  

Has the change to RTF in the Script Editor been documented anywhere?

**Comment by SeriousBuggie** [comment]
Try paste it to some good text edit which not automatically convert line ending and view file in HEX editor.

**Comment by OjitroC** [comment]
Thanks for the idea - however I only have the Mac's default app TextEdit (basically because I don't need anything else) and, as far as I know, I don't have a HEX editor. 

The issue seems to be that the 469b Script Editor only accepts RTF whilst the Script Editor in 436 accepts both plain text and RTF.  So it comes down to - is this intentional?  Is it an error?  If it is intentional then is it necessary? If so, why and perhaps it should be mentioned somewhere?

**Comment by stijn-volckaert** [comment]
I have no issues pasting plain text into the code editor. The text shows up with the expected line endings. 

We made no functional changes to the editor that would affect copy-pasting behavior except that we load msftedit.dll instead of riched20.dll now, as recommended by MSDN.

Based on your description, it sounds like wine's implementation of msftedit behaves differently than Windows' implementation. Do you have any details on your setup that could help me debug this on one of my own mac machines (wine version, configuration, instructions on how to use wineskins on mac, ...). I have never tried to run Wine on a Mac before...

**Comment by OjitroC** [comment]
Thanks for the offer to debug the issue.  I'm not sure it is really worth all the effort for you that this would entail.  The Winskin and Windows versions are quite old - they are
Wrapper version : WIneskin 2.6.0
Engine : WS9Wine 1.7.28
Windows 5.1.2600

I had UT451 and WineSkin in a complete package from The Unrealism Zone (the Mac OS UT site) - this is now in the process of closing down and downloads are no longer available.  I moved UT 436 from my old Windows machine and can now run this, 451 and 469b on my iMac using the Wineskin wrapper.

I have no idea how to set it all up as I simply ran the installer with the package.

I can paste RTF text into the 469b Script Editor without any issues and can paste plain text into the 436 Script Editor again without issues - so this issue is not really that big a deal.
